### PR TITLE
[ActionList][Item] Allow indentation to be applied to individual items

### PR DESCRIPTION
**Pull request by zakwarsame**
### WHY are these changes introduced?

Fixes #0000 

Fixes https://github.com/Shopify/polaris-summer-editions/issues/785
Fixes https://github.com/Shopify/core-workflows/issues/1029
Fixes https://github.com/Shopify/polaris-summer-editions/issues/789

<details>
   <summary> Expand to see specifications/designs for this PR: </summary>

image

</details>

### WHAT is this pull request doing?

- This PR takes an alternative approach to [this PR](https://github.com/Shopify/polaris/pull/9771)
- It adds a new prop to `ActionList`'s `Item` component called `indent`. This prop adds some space and a border to the left of the individual items.
- Given that the indentation only applies to `Plus` stores in the original design, we can simply specify the prop in the consumer.

### How to 🎩

Non-Plus: https://admin.web.mn.zakaria-warsame.us.spin.dev/store/shop1
Plus: https://admin.web.mn.zakaria-warsame.us.spin.dev/store/shop2

Ensure the user menu matches the designs

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
 ~- [ ] Updated the component's `README.md` with documentation changes~
- [x] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-react/src/components/ActionList/components/Section/Section.tsx [hunk 1]
```diff
@@ -110,13 +120,17 @@ export function Section({
...
-        as="ul"
+        as={polarisSummerEditions2023 ? 'div' : 'ul'}
         padding={polarisSummerEditions2023 ? '1_5-experimental' : '2'}
         {...(hasMultipleSections && {paddingBlockStart: '0'})}
         {...(sectionRole && {role: sectionRole})}
         tabIndex={!hasMultipleSections ? -1 : undefined}
       >
-        {actionMarkup}
+        {polarisSummerEditions2023 ? (
+          <VerticalStack gap="1">{actionMarkup}</VerticalStack>
```
Comment (zakwarsame) [inline polaris-react/src/components/ActionList/components/Section/Section.tsx]: Outside indentation, this is the change from our designs that will affect all items. It provide a gap to list items:


<details>
  <summary>Before </summary>

Screenshot 2023-07-24 at 9 42 21 PM


</details>

<details>
  <summary>After </summary>

Screenshot 2023-07-24 at 9 41 44 PM


</details>

File: .changeset/angry-mirrors-explode.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': minor
+---
+
+Added a prop to indent items to ActionList
```
Comment (sam-b-rose) [inline .changeset/angry-mirrors-explode.md]:
```suggestion
Added an `indent` prop to ActionList for applying left indentation space and border to items
```

File: polaris-react/src/components/TopBar/TopBar.stories.tsx [hunk 3]
```diff
@@ -1,7 +1,13 @@
 import React, {useCallback, useState} from 'react';
 import type {ComponentMeta} from '@storybook/react';
 import {ActionList, Frame, Icon, TopBar, Text, Avatar} from '@shopify/polaris';
-import {ArrowLeftMinor, QuestionMarkMajor} from '@shopify/polaris-icons';
+import {
+  ArrowLeftMinor,
+  QuestionMarkMajor,
+  ResetMinor,
```
Comment (sam-b-rose) [inline polaris-react/src/components/TopBar/TopBar.stories.tsx]: Are these added icon imports left over from previous work?

**Review by sam-b-rose — APPROVED** [review]
This approach looks excellent! Very composable and simple API for toggling the indentation style.

The examples in the spin link look good. I just noticed one spacing issue where the selected Store has a different horizontal margin than the other items. Though I'm not sure if that is an issue within Polaris or in the consumer implementation

Screenshot 2023-07-25 at 7 57 29 AM

**Review by kyledurand — APPROVED** [review]
👏

**Comment by chrisblinstrub** [comment]
Hey @zakwarsame, not sure if these items are relevant to this specific PR (apologies if they aren't!), but I noticed three things:

1. In the Non-Plus link, the store icons for the non-active store are 20x20 when they should be 24x24. You'll see how Shop 3 and Shop 4 are further to the left than Shop 1:



2. In the Non-Plus link, the icon for the "All Stores" link is sitting in a 20x20 container when it should be 24x24. This is causing similar alignment issues as above screenshot. 



3. This may be the same thing as item 2, but in the Plus link the icon for the "Switch organization" link is sitting in a 20x20 container when it should be 24x24.

File: polaris-react/src/components/TopBar/TopBar.stories.tsx (hunk 3 above)
Comment (zakwarsame) [reply polaris-react/src/components/TopBar/TopBar.stories.tsx]: Yup, good catch 😅

**Review by sam-b-rose — COMMENTED** [review]
🚢 👍

**Pull request merged**